repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/via/ViaHeaderOldAttributeConfigTest.groovy

```diff
@@ -0,0 +1,200 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+
+package features.core.via
+
+import framework.ReposeValveTest
+import framework.server.CustomizableSocketServerConnector
+import org.apache.http.HttpVersion
+import org.apache.http.client.HttpClient
+import org.apache.http.client.methods.HttpGet
+import org.apache.http.client.methods.HttpUriRequest
+import org.apache.http.impl.client.HttpClients
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.Endpoint
+import org.rackspace.deproxy.MessageChain
+import org.rackspace.deproxy.Response
+import spock.lang.Shared
+import spock.lang.Unroll
+
+import static framework.server.CustomizableSocketServerConnector.HTTP_1_0
+import static framework.server.CustomizableSocketServerConnector.HTTP_1_1
+import static javax.servlet.http.HttpServletResponse.SC_OK
+import static org.rackspace.deproxy.Deproxy.REQUEST_ID_HEADER_NAME
+import static org.springframework.http.HttpHeaders.VIA
+
+/**
+ * Tests the contents of the Via header when it's configured using the old "via" attribute in the "deployment-config"
+ * element in the container config.
+ */
+class ViaHeaderOldAttributeConfigTest extends ReposeValveTest {
+    private static final String VIA_PREFIX = "Cactuar"
+
+    @Shared String reposePort
+    @Shared String reposeVersion
+    @Shared CustomizableSocketServerConnector socketServerConnector
+
+    def setupSpec() {
+        deproxy = new Deproxy()
+        Endpoint originService = deproxy.addEndpoint(name: 'origin service', connectorFactory: { Endpoint endpoint ->
+            new CustomizableSocketServerConnector(endpoint, properties.targetPort)
+        })
+        socketServerConnector = originService.serverConnector as CustomizableSocketServerConnector
```

Comment:
I'm fetching the output of the closure after Deproxy has fed it an Endpoint.

The alternative is this:
```
deproxy.addEndpoint(name: 'origin service', connectorFactory: { Endpoint endpoint ->
    socketServerConnector = new CustomizableSocketServerConnector(endpoint, properties.targetPort)
    socketServerConnector
})
```